### DISCUSSION TITLE
Scraping Kubernetes etcd, controller-manager, scheduler

### DISCUSSION THREAD

**Discussion by kyle-rf** [prometheus-operator/kube-prometheus#1121] [Q&A] [answered]
I'm not sure if I should have this here or in the prometheus-community/helm-charts, but they don't have discussions enabled.  

I have the kube-prometheus-stack helm chart deployed in my environment.  In order to get Prometheus to scrape etcd, kube-controller-manager, and kube-scheduler I modified the Kubernetes manifest yaml files for those resources.  I had to modify bind_address and the metrics address to be 0.0.0.0.  Which were originally set to 127.0.0.1.

etcd: 
![image](https://user-images.githubusercontent.com/26102836/116570265-c020be00-a8cf-11eb-9f1a-5e40ef7a41e0.png)

scheduler and controller-manager:
![image](https://user-images.githubusercontent.com/26102836/116570585-00803c00-a8d0-11eb-941a-dc626de0d39c.png)

When I run an upgrade of the Kubernetes cluster the control-plane nodes update their manifests and reset these values back to 127.0.0.1.  First off: I am not sure if this is problematic or if there are security concerns for switching these from localhost or if there is a better approach to being able to scrape these.  It is just the way I was able to get those components scraped.  I'd rather not have to go and change this every time if possible.  So long as there are not any concerns to having it set this way it isn't too much trouble I guess.

I made this post so if anyone has any insight on whether this is a bad approach on my part or knows of a better way I would appreciate the feedback.

**Comment by paulfantom** [comment, marked as answer]
Neither helm chart nor kube-prometheus project will configure kube-scheduler, kube-controller-manager, nor etcd. This is because there are multiple ways of doing this which depend on the platform and installer you used to setup your kubernetes cluster. Due to myriad of possibilities, it is usually better to describe it in the documentation. We do have a guide about how to do this on kubeadm, but sadly not for other platforms: https://github.com/prometheus-operator/kube-prometheus/blob/main/docs/kube-prometheus-on-kubeadm.md. We are open to contributing new guides for other platforms!

**Reply by kyle-rf** [reply]
Thank you @paulfantom.  I will look over the docs you referenced.
